experimental/wazerotest/wazerotest.go

```diff
@@ -0,0 +1,605 @@
+	ExportNames  []string
+
+	// Lazily initialized when accessed through the module.
+	module *Module
+	index  int
+}
+
+var (
+	errMissingFunctionSignature      = errors.New("missing function signature")
+	errMissingFunctionModule         = errors.New("missing function module")
+	errMissingFunctionImplementation = errors.New("missing function implementation")
+)
+
+func (f *Function) Definition() api.FunctionDefinition {
+	return functionDefinition{function: f}
+}
+
+func (f *Function) Call(ctx context.Context, params ...uint64) ([]uint64, error) {
+	stackLen := len(f.ParamTypes)
+	if stackLen < len(f.ResultTypes) {
+		stackLen = len(f.ResultTypes)
+	}
+	stack := make([]uint64, stackLen)
+	copy(stack, params)
+	err := f.CallWithStack(ctx, stack)
+	if err != nil {
+		for i := range stack {
+			stack[i] = 0
+		}
+	}
+	return stack[:len(f.ResultTypes)], err
+}
+
+func (f *Function) CallWithStack(ctx context.Context, stack []uint64) error {
+	if f.ParamTypes == nil || f.ResultTypes == nil {
+		return errMissingFunctionSignature
+	}
+	if f.GoModuleFunction == nil {
+		return errMissingFunctionImplementation
+	}
+	if f.module == nil {
+		return errMissingFunctionModule
+	}
+	if exitCode, exited := f.module.ExitStatus(); exited {
+		return sys.NewExitError(exitCode)
+	}
+	f.GoModuleFunction.Call(ctx, f.module, stack)
+	return nil
+}
+
+type functionDefinition struct {
+	internalapi.WazeroOnlyType
+	function *Function
+}
+
+func (def functionDefinition) Name() string {
+	return def.function.FunctionName
+}
+
+func (def functionDefinition) DebugName() string {
+	if def.function.DebugName != "" {
+		return def.function.DebugName
+	}
+	return fmt.Sprintf("%s.$%d", def.ModuleName(), def.Index())
+}
+
+func (def functionDefinition) GoFunction() any {
+	return def.function.GoModuleFunction
+}
+
+func (def functionDefinition) ParamTypes() []api.ValueType {
+	return def.function.ParamTypes
+}
+
+func (def functionDefinition) ParamNames() []string {
+	return def.function.ParamNames
+}
+
+func (def functionDefinition) ResultTypes() []api.ValueType {
+	return def.function.ResultTypes
+}
+
+func (def functionDefinition) ResultNames() []string {
+	return def.function.ResultNames
+}
+
+func (def functionDefinition) ModuleName() string {
+	if def.function.module != nil {
+		return def.function.module.ModuleName
+	}
+	return ""
+}
+
+func (def functionDefinition) Index() uint32 {
+	return uint32(def.function.index)
+}
+
+func (def functionDefinition) Import() (moduleName, name string, isImport bool) {
+	return
+}
+
+func (def functionDefinition) ExportNames() []string {
+	return def.function.ExportNames
+}
+
+// Memory is an implementation of the api.Memory interface, representing the
+// memory of a WebAssembly module.
+type Memory struct {
+	internalapi.WazeroOnlyType
+
+	// Byte slices holding the memory pages.
+	//
+	// It is the user's repsonsibility to ensure that the length of this byte
+	// slice is a multiple of the page size.
+	Pages []byte
```

Comment:
Pages and also "Byte slices" makes me think it should be `[][65536]byte`. Maybe `Raw` or `Buffer` is better name? Whatever we do I think it shouldn't be named differently than our default impl